pkg/actuators/machine/machine_scope.go

```diff
@@ -208,6 +208,12 @@ func (s *machineScope) getUserData() ([]byte, error) {
 		return nil, fmt.Errorf("The userData secret %s missing %s key", objKey, userDataSecretKey)
 	}
 
+	// WMCO already takes case of setting the hostname to the same as the Machine name. So just return userData.
+	if s.machine.Labels["machine.openshift.io/os-id"] == "Windows" {
+		klog.V(3).Infof("%s: The windows-user-data: %s", s.machine.Name, string(userData))
```

Comment:
is there a possibility that we could leak sensitive information by printing the user data in the logs?

Reply:
The windows-user-data is just the powershell script for the VM initialization, just as the ignition data of rhcos. I think there is no sensitive data such as the credentials or password, etc. in it.

Reply:
We also logs the rhcos ignition user-data in this function.

Reply:
We avoid printing the user data in our logs to be on the safe side as it does contain the public key. While this information is not sensitive we felt it adds no debugging value.